module/VuFind/src/VuFind/Db/Table/Resource.php

```diff
@@ -204,14 +204,12 @@ function ($s) use ($user, $list, $tags, $sort, $offset, $limit, $obj) {
 
                 // Adjust for tags if necessary:
                 if (!empty($tags)) {
-                    $linkingTable = $obj->getDbTable('ResourceTags');
+                    $linkingTable = $this->getDbService(\VuFind\Db\Service\TagService::class);
                     foreach ($tags as $tag) {
                         $matches = $linkingTable
-                            ->getResourcesForTag($tag, $user, $list)->toArray();
-                        $getId = function ($i) {
-                            return $i['resource_id'];
-                        };
-                        $s->where->in('resource.id', array_map($getId, $matches));
+                            ->getResourcesForTag($tag, $user, $list);
```

Comment:
Is this method really just returning an array of IDs instead of an array of objects? If so, I wonder if it would make sense to rename it to `getResourceIDsForTag` to clarify the nature of the return value.